File: backend/src/main/java/com/backend/api/resume/dto/response/ResumeCreateResponse.java
package com.backend.api.resume.dto.response;

import com.backend.domain.resume.entity.Resume;
import com.backend.domain.user.entity.User;
import io.swagger.v3.oas.annotations.media.Schema;

public record ResumeCreateResponse(
        @Schema(description = "이력서 ID", example = "1")
        Long resumeId,
        @Schema(description = "사용자 ID", example = "1")
        Long userId,
        @Schema(description = "이력서 내용", example = "이력서 내용입니다.")
        String content,
        @Schema(description = "기술 스택", example = "Java, Spring Boot")
        String skill,
        @Schema(description = "대외 활동", example = "대외 활동 내용입니다.")
        String activity,
        @Schema(description = "자격증", example = "없음")
        String certification,
        @Schema(description = "경력 사항", example = "경력 사항 내용입니다.")
        String career,
        @Schema(description = "포트폴리오 URL", example = "http://portfolio.example.com")
        String portfolioUrl
) {
    public static ResumeCreateResponse from(Resume resume, User user) {
        return new ResumeCreateResponse(
                resume.getId(),
                user.getId(),
                resume.getContent(),
                resume.getSkill(),
                resume.getActivity(),
                resume.getCertification(),
                resume.getCareer(),
                resume.getPortfolioUrl()
        );
    }
}

File: backend/src/main/java/com/backend/api/resume/dto/response/ResumeCreateResponse.kt
package com.backend.api.resume.dto.response

import com.backend.domain.resume.entity.Resume
import com.backend.domain.user.entity.User
import io.swagger.v3.oas.annotations.media.Schema


data class ResumeCreateResponse(
    @field:Schema(
        description = "이력서 ID",
        example = "1"
    ) val resumeId: Long,
    @field:Schema(
        description = "사용자 ID",
        example = "1"
    ) val userId: Long,
    @field:Schema(
        description = "이력서 내용",
        example = "이력서 내용입니다."
    ) val content: String?,
    @field:Schema(
        description = "기술 스택",
        example = "Java, Spring Boot"
    ) val skill: String?,
    @field:Schema(
        description = "대외 활동",
        example = "대외 활동 내용입니다."
    ) val activity: String?,
    @field:Schema(
        description = "자격증",
        example = "없음"
    ) val certification: String?,
    @field:Schema(
        description = "경력 사항",
        example = "경력 사항 내용입니다."
    ) val career: String?,
    @field:Schema(
        description = "포트폴리오 URL",
        example = "http://portfolio.example.com"
    ) val portfolioUrl: String?
) {
    companion object {
        fun from(resume: Resume, user: User): ResumeCreateResponse {
            return ResumeCreateResponse(
                resume.id,
                user.id,
                resume.content,
                resume.skill,
                resume.activity,
                resume.certification,
                resume.career,
                resume.portfolioUrl
            )
        }
    }
}
